src/main/scala/swiss/dasch/api/ExportEndpoint.scala

```diff
@@ -0,0 +1,72 @@
+package swiss.dasch.api
+
+import swiss.dasch.domain.{ AssetService, ProjectShortcode }
+import zio.{ ZIO, ZNothing, http }
+import zio.http.HttpError.*
+import zio.http.*
+import zio.http.codec.HttpCodec
+import zio.http.codec.HttpCodec.*
+import zio.http.endpoint.{ Endpoint, EndpointMiddleware, Routes }
+import zio.http.endpoint.EndpointMiddleware.None
+import zio.json.{ DeriveJsonEncoder, JsonEncoder }
+import zio.nio.file
+import zio.schema.{ DeriveSchema, Schema }
+import zio.schema.DeriveSchema.gen
+
+object ExportEndpoint {
+  final case class ExportResponse(path: String)
+  private object ExportResponse {
+    def make(path: zio.nio.file.Path): ExportResponse = ExportResponse(path.toString)
+
+    implicit val encoder: JsonEncoder[ExportResponse] = DeriveJsonEncoder.gen[ExportResponse]
+    implicit val schema: Schema[ExportResponse]       = DeriveSchema.gen[ExportResponse]
+  }
+
+  private val shortcodePathVar = "shortcode"
+
+  private val exportEndpoint: Endpoint[String, ApiProblem, ExportResponse, None] = Endpoint
+    .post("export" / string(shortcodePathVar))
+    .out[ExportResponse](Status.Ok)
+    .outErrors(
+      HttpCodec.error[ProjectNotFound](Status.NotFound),
+      HttpCodec.error[IllegalArguments](Status.BadRequest),
+      HttpCodec.error[InternalProblem](Status.InternalServerError),
+    )
+
+  private val exportRoute: Routes[AssetService, ApiProblem, None] = exportEndpoint.implement { (shortcode: String) =>
+    ZIO
+      .fromEither(ProjectShortcode.make(shortcode))
+      .mapError(ApiProblem.invalidPathVariable(shortcodePathVar, shortcode, _))
+      .flatMap(AssetService.zipProject(_).some.mapError {
+        case Some(e) => ApiProblem.internalError(e)
+        case _       => ProjectNotFound(shortcode)
+      })
+      .map(ExportResponse.make)
+  }
+
+  val app = exportRoute.toApp
+
+  sealed trait ApiProblem
+  case class ProjectNotFound(shortcode: String)            extends ApiProblem
+  case class IllegalArguments(errors: Map[String, String]) extends ApiProblem
+  case class InternalProblem(errorMessage: String)         extends ApiProblem
+
+  object ApiProblem {
```

Comment:
I am inspired by the `ApiProblem` RFC https://www.rfc-editor.org/rfc/rfc7807

The current implementation does not really comply with that standard but it could worth thinking about implementing the parts we think are useful for us.

Reply:
I wasn't aware of that - seems interesting at first glance. In that case, the naming makes sense of course